I've got a failure on ttl.sh in a PR:
https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/30934642785/job/92089747868?pr=5730

```
Pushing signature to: ttl.sh/0191a076b6b65daf32f683be5e42ea4d05abc8b3/traceloop
Error: signing [ttl.sh/0191a076b6b65daf32f683be5e42ea4d05abc8b3/traceloop@sha256:daef5a855c313adfb6722ce0c4b8e89b1254a8b95f9775687beba6eee2624d82]: recursively signing: signing digest: Patch "https://ttl.sh/v2/0191a076b6b65daf32f683be5e42ea4d05abc8b3/traceloop/blobs/uploads/9115837a-ef57-48b7-97df-c226bf0d80b7": context deadline exceeded
error during command execution: signing [ttl.sh/0191a076b6b65daf32f683be5e42ea4d05abc8b3/traceloop@sha256:daef5a855c313adfb6722ce0c4b8e89b1254a8b95f9775687beba6eee2624d82]: recursively signing: signing digest: Patch "https://ttl.sh/v2/0191a076b6b65daf32f683be5e42ea4d05abc8b3/traceloop/blobs/uploads/9115837a-ef57-48b7-97df-c226bf0d80b7": context deadline exceeded
make: *** [Makefile:175: traceloop-sign-existing] Error 1
```

I think you're right, I will remove the `-j` unconditionally.

--- a/.github/workflows/inspektor-gadget.yml
+++ b/.github/workflows/inspektor-gadget.yml
@@ -1952,6 +1952,7 @@ jobs:
           BUILDER_IMAGE: ${{ needs.build-helper-images.outputs.gadget_builder_image }}
           GADGET_REPOSITORY: ${{ steps.set-repo-determine-image-tag.outputs.gadget-repository }}
           GADGET_TAG: ${{ steps.set-repo-determine-image-tag.outputs.gadget-tag }}
+          GADGET_PUSH_RETRIES: 5
           CFLAGS: "-Werror"
         run: |
           make build-gadgets -o install/ig -j$(nproc)
@@ -1960,7 +1961,7 @@ jobs:
           git diff --exit-code HEAD --
 
           # Avoid building the gadgets again
-          make -C gadgets/ push-existing -j$(nproc)
+          make -C gadgets/ push-existing
       - name: Sign the gadgets
         if: needs.check-secrets.outputs.cosign == 'true'
         env:
@@ -1968,8 +1969,9 @@ jobs:
           COSIGN_PRIVATE_KEY: '${{ secrets.COSIGN_PRIVATE_KEY }}'
           GADGET_REPOSITORY: ${{ steps.set-repo-determine-image-tag.outputs.gadget-repository }}
           GADGET_TAG: ${{ steps.set-repo-determine-image-tag.outputs.gadget-tag }}
+          GADGET_PUSH_RETRIES: 5
         run: |
-          make -C gadgets/ sign-existing -j$(nproc)
+          make -C gadgets/ sign-existing
 
   gadgets-unittest:
     name: Gadgets unit tests

--- a/gadgets/Makefile
+++ b/gadgets/Makefile
@@ -14,6 +14,11 @@ KUBECTL_GADGET ?= kubectl-gadget
 GPU_EBPF_BRIDGE ?= gpu-ebpf-bridge
 IG_RUNTIME ?= docker
 IG_FLAGS ?=
+# Number of times to retry pushing/signing a gadget after a failure.
+# Defaults to 0 (a single attempt) so interactive users are not slowed
+# down; CI sets this to work around flaky ghcr.io 500s on concurrent
+# manifest PUTs.
+GADGET_PUSH_RETRIES ?= 0
 IG_DEBUG_LOGS ?= true
 COSIGN ?= cosign
 # Force using legacy .sig format.
@@ -158,7 +163,16 @@ push: $(addsuffix -push,$(GADGETS))
 
 %-push-existing: %
 	@echo "Pushing existing $*"
-	@sudo -E $(IG) image push $(GADGET_REPOSITORY)/$*:$(GADGET_TAG) $(IG_FLAGS)
+	@n=0; \
+	until sudo -E $(IG) image push $(GADGET_REPOSITORY)/$*:$(GADGET_TAG) $(IG_FLAGS); do \
+		if [ $$n -ge $(GADGET_PUSH_RETRIES) ]; then \
+			echo "Push of $* failed after $$n retries"; \
+			exit 1; \
+		fi; \
+		n=$$((n+1)); \
+		echo "Push of $* failed, retry $$n/$(GADGET_PUSH_RETRIES) in $$((n*5))s..."; \
+		sleep $$((n*5)); \
+	done
 
 .PHONY:
 push-existing: $(addsuffix -push-existing,$(GADGETS))
@@ -172,8 +186,17 @@ sign: $(addsuffix -sign,$(GADGETS))
 
 %-sign-existing:
 	@echo "Signing existing $*"
-	digest=$$(sudo -E $(IG) image list --no-trunc | grep "$* " | awk '{ print $$3 }') ; \
-	$(COSIGN) sign $(COSIGN_FLAGS) --key env://COSIGN_PRIVATE_KEY --yes --recursive $(GADGET_REPOSITORY)/$*@$$digest
+	@digest=$$(sudo -E $(IG) image list --no-trunc | grep "$* " | awk '{ print $$3 }') ; \
+	n=0; \
+	until $(COSIGN) sign $(COSIGN_FLAGS) --key env://COSIGN_PRIVATE_KEY --yes --recursive $(GADGET_REPOSITORY)/$*@$$digest; do \
+		if [ $$n -ge $(GADGET_PUSH_RETRIES) ]; then \
+			echo "Signing of $* failed after $$n retries"; \
+			exit 1; \
+		fi; \
+		n=$$((n+1)); \
+		echo "Signing of $* failed, retry $$n/$(GADGET_PUSH_RETRIES) in $$((n*5))s..."; \
+		sleep $$((n*5)); \
+	done
 
 sign-existing: $(addsuffix -sign-existing,$(GADGETS))
 

